src/ast/variables/NamespaceVariable.ts

```diff
@@ -77,32 +83,45 @@ export default class NamespaceVariable extends Variable {
 		const _ = options.compact ? '' : ' ';
 		const n = options.compact ? '' : '\n';
 		const t = options.indent;
+		const syntheticNamedExports = this.syntheticNamedExports;
 
-		const members = Object.keys(this.memberVariables).map(name => {
-			const original = this.memberVariables[name];
+		const members = Object.keys(this.memberVariables)
+			.filter(name => !syntheticNamedExports || name !== 'default')
```

Comment:
I do not think you should skip the default export. It still exists and should be part of the namespace.

Reply:
I wrote some comments about this earlier, withoyt this we will always generate:
```
const ns = Object.assign({default: a}, a);
```

When you use `syntheticNamedExports`... the whole namespace is the default export, so there is not really a default, unless the default export, has a "default" property on it.

Reply:
That is unfortunately not the case if you follow proper ESM semantic. Which I think we should, otherwise our code could correctly be considered buggy. Consider this example:

https://rollupjs.org/repl/?version=1.27.13&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMColMjBhcyUyMGRlcCUyMGZyb20lMjAnLiUyRmRlcC5qcyclM0IlNUNuJTVDbmZ1bmN0aW9uJTIwbG9nRGVmYXVsdChucyklMjAlN0IlNUNuJTIwJTIwY29uc29sZS5sb2cobnMuZGVmYXVsdCklM0IlNUNuJTdEJTVDbiU1Q25sb2dEZWZhdWx0KGRlcCklM0IlMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyZGVwLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGRlZmF1bHQlMjAnY29ycmVjdCclM0IlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXNtJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

Following ES module namespace semantics, this example should log (and does in fact log) `correct`.  With your change, it will suddenly log `undefined` because the default export is suppressed and the module is broken.

So yes, I think this is exactly what should happen, you should always use `Object.assign` when synthetic named exports are activated, because otherwise it will no longer follow ES module semantics and possibly break code.

Reply:
Note though that "reified" namespaces are not very common as people usually do not directly pass namespaces around but rather access their properties, in which case Rollup will generate direct references. And in case such a namespace is generated, the copying will only happen at most once per module which I think should not be a too big performance hit.

Reply:
I hope for the next major version to switch to generating ES6 code anyways, in which case we can replace the object assign with a spread operator, which should be more efficient in all regards.